test/unit/alerts_test.rb

```diff
@@ -46,4 +49,69 @@ def setup
 
     assert_true(Alerts.can_raise_more_alerts?(@service_id, @application_id))
   end
+
+  # Tests for Alerts::UsagesChecked start here
+
+  test 'when marking as checked, it also sets the key with the minimal TTL among the notified bins' do
+    min_ttl = 60
+
+    key_bin_50 = Alerts.send(:key_already_notified, @service_id, @application_id, 50)
+    @storage.setex(key_bin_50, min_ttl*2, '1')
+
+    key_bin_80 = Alerts.send(:key_already_notified, @service_id, @application_id, 80)
+    @storage.setex(key_bin_80, min_ttl, '1')
+
+    Alerts::UsagesChecked.mark_all_checked(@service_id, @application_id)
+
+    assert_false Alerts::UsagesChecked.need_to_check_all?(@service_id, @application_id)
+
+    # The 80% bin has a lower TTL. We cannot guarantee the exact TTL because
+    # some time might pass between the moment we create the key and check its
+    # TTL, but we know it needs to be <= min_ttl and close to it.
```

Comment:
So maybe just ensure the offset is something bigger than a time difference since before you create the key and right before you check its TTL to reduce the odds of spurious failures?

Reply:
I think that's difficult to get exactly right, that's why I used a range instead.